hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestLocalProcessLockProvider.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestLocalProcessLockProvider {
+
+  private static final Logger LOG = LogManager.getLogger(TestLocalProcessLockProvider.class);
+  private final Configuration hadoopConfiguration = new Configuration();
+  private final LockConfiguration lockConfiguration = new LockConfiguration(new TypedProperties());
+
+  @Test
+  public void testLockAcquisition() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testLockReAcquisitionBySameThread() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+    assertThrows(HoodieLockException.class, () -> {
+      localProcessLockProvider.lock();
+    });
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testLockReAcquisitionByDifferentThread() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+
+    // Main test thread
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+
+    // Another writer thread
+    Thread writer2 = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        assertThrows(HoodieLockException.class, () -> {
+          localProcessLockProvider.lock();
+        });
+      }
+    });
+
+    try {
+      writer2.join();
+    } catch (InterruptedException e) {
+      //
+    }
+
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testTryLockAcquisition() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    Assertions.assertTrue(localProcessLockProvider.tryLock());
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testTryLockAcquisitionWithTimeout() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    Assertions.assertTrue(localProcessLockProvider.tryLock(1, TimeUnit.MILLISECONDS));
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testTryLockReAcquisitionBySameThread() {
```

Comment:
can we add a test where in, write1 acquires the lock. write2 try to acquire with 1 sec(or 5 secs). and within 1sec, writer1 releases the lock and writer2 was able to successfully acquire the lock.

Reply:
Added a new unit test for your suggested case.